src/typing.py

```diff
@@ -1985,20 +1987,29 @@ class Employee(NamedTuple):
         The resulting class has one extra attribute: _field_types,
         giving a dict mapping field names to types.  (The field names
         are in the _fields attribute, which is part of the namedtuple
-        API.) Backward-compatible usage::
+        API.) Alternative equivalent keyword syntax is also accepted::
+
+            Employee = NamedTuple('Employee', name=str, id=int)
+
+        Backward-compatible usage::
 
             Employee = NamedTuple('Employee', [('name', str), ('id', int)])
         """
 
-        def __new__(self, typename, fields):
+        def __new__(self, typename, fields=None, **kwargs):
+            if fields is None:
+                fields = kwargs.items()
```

Comment:
@gvanrossum This code is also inside ``if sys.version_info[:2] >= (3, 6):`` I mentioned above. So that the ``else:`` branch is used for other versions, and both keyword and class syntax fail with a ``TypeError``.

I will add a new commit now, where I will "unify" the code, this will give better error messages.